Tests/CacheAdvanceTests/SQLitePerformanceComparisonTests.swift

```diff
@@ -0,0 +1,307 @@
+            maxMessageCount: Int32(LorumIpsum.messages.count),
+            shouldOverwriteMessages: true)
+        for message in LorumIpsum.messages {
+            cache.appendMessage(message)
+        }
+
+        XCTAssertEqual(cache.messages(), LorumIpsum.messages)
+    }
+
+    // MARK: Performance Tests
+
+    func test_performance_createDatabaseAndAppendSingleMessage() {
+        measure {
+            // Delete any existing database.
+            try? FileManager.default.removeItem(at: testFileLocation)
+
+            let cache = SQLiteCache<TestableMessage>(
+                location: testFileLocation,
+                maxMessageCount: Int32(LorumIpsum.messages.count),
+                shouldOverwriteMessages: false)
+
+            cache.appendMessage("test message")
+        }
+    }
+
+    func test_performance_sqlite_append_fillableCache() {
+         // Create a cache that won't run out of room over multiple test runs.
+        let cache = SQLiteCache<TestableMessage>(
+            location: testFileLocation,
+            maxMessageCount: Int32(LorumIpsum.messages.count * 10),
+            shouldOverwriteMessages: false)
+        cache.createDatabaseIfNecessay()
+
+        measure {
+            for message in LorumIpsum.messages {
+                cache.appendMessage(message)
+            }
+        }
+    }
+
+    func test_performance_sqlite_append_overwritingCache() {
+        let cache = SQLiteCache<TestableMessage>(
+            location: testFileLocation,
+            maxMessageCount: Int32(LorumIpsum.messages.count),
+            shouldOverwriteMessages: true)
+
+        // Fill the cache before the test starts.
+        for message in LorumIpsum.messages {
+            cache.appendMessage(message)
+        }
+        measure {
+            for message in LorumIpsum.messages {
+                cache.appendMessage(message)
+            }
+        }
+    }
+
+    func test_performance_sqlite_messages_fillableCache() {
+        let cache = SQLiteCache<TestableMessage>(
+            location: testFileLocation,
+            maxMessageCount: Int32(LorumIpsum.messages.count),
+            shouldOverwriteMessages: false)
+        for message in LorumIpsum.messages {
+            cache.appendMessage(message)
+        }
+        measure {
+            _ = cache.messages()
+        }
+    }
+
+    func test_performance_sqlite_messages_overwritingCache() {
+        let cache = SQLiteCache<TestableMessage>(
+            location: testFileLocation,
+            maxMessageCount: Int32(LorumIpsum.messages.count),
+            shouldOverwriteMessages: true)
+        for message in LorumIpsum.messages + LorumIpsum.messages {
+            cache.appendMessage(message)
+        }
+        measure {
+            _ = cache.messages()
+        }
+    }
+
+    // MARK: Private
+
+    private let testFileLocation = FileManager.default.temporaryDirectory.appendingPathComponent("SQLiteTests")
+}
+
+class SQLiteCache<T: Codable> {
```

Comment:
To be honest I didn't fully process the SQLite code since we have test coverage that it does what we expect.